### ISSUE TITLE
deprecation of dub fetch package --version causes problems for IDEs that call dub

### ISSUE THREAD

**Issue by SingingBush** [dlang/dub#2028]
the [intellij-dlanguage](https://github.com/intellij-dlanguage/intellij-dlanguage/) plugin makes use of `dub fetch <package> --version=<version>` but now that the `--version` arg is deprecated there's a potential problem for projects such as ours. 

I found #1557 and some related commits. Is the correct thing for our plugin to do to use `dub fetch <package>@<version>`?

We'd need to make major changes to have the code use different syntax for versions of dub up to the point that it changed and a different syntax for versions after. So hopefully just switching to from `--version=<version>` to `<package>@<version>` will result in the same behaviour **and work on all versions of dub**.

please confirm if this is the case.

**Comment by wilzbach** [comment]
See https://github.com/dlang/dub/pull/1987 for details.

tl;dr: `<package>@<version>` is the newer syntax.

> and work on all versions of dub.

No, only >= 1.23.

**Comment by SingingBush** [comment]
if it's only for dub >= 1.23 please continue to support `dub fetch <package> --version=<version>`

**Comment by wilzbach** [comment]
Isn't it is still supported?
This is just about the deprecation message then?

**Comment by SingingBush** [comment]
the deprecation isn't a problem in itself. It's the pending removal that I'm concerned about. The reason for creating the issue was to let you know that my project relies on it and request that it doesn't get removed too soon.

**Comment by wilzbach** [comment]
Ok. Understood. Thanks a lot for letting us know. We generally try to keep things around for at least two years with DMD or Phobos, so it would only make sense to do so here as well.
Additionally, there are currently no plans for dub v2, so technically we can't even introduce a breaking change.
Anyhow, I will keep an eye on it.

**Comment by Geod24** [comment]
@SingingBush : The long term plan is to remove it, but indeed we keep a compatibility of 10 releases (usually 2 years).
With the most recent work going into DUB, we probably will postpone any removal for a dub v2.0.
